### PR TITLE
SolidJS Linting prettier

### PR DESCRIPTION
**Pull request by hdJerry**
# Background
Having code style consistency is important for any project. A standardized set of options should be set that can be overwritten by the implementer at a later date.

# Acceptance
- [x] Establish eslint/tslint rules and configuration
- [x] Install and configure prettier
- [x] Configure editorconfig settings
- [x] Run linting and prettier on entire project and correct any errors

# NOTE:
This task has already been completed many times on different KITs, so we should make sure to look at those configurations first and try not to reinvent the wheel :) 

Screenshot 2022-10-18 at 13 54 55
Screenshot 2022-10-18 at 13 55 26

### CODE REVIEW

File: starters/solidjs-tanstack-tailwind/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Editor configuration, see https://editorconfig.org
+root = true
+
+[*]
+charset = utf-8
```
Comment (mbicknese) [inline starters/solidjs-tanstack-tailwind/.editorconfig:5]: editorconfig also allows us to set `end_of_line`. Shall we add that with the value of "lf"?

File: starters/solidjs-tanstack-tailwind/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+indent_style = space
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.jsx]
+quote_type = single
```
Comment (mbicknese) [inline starters/solidjs-tanstack-tailwind/.editorconfig:12]: Does this work? It doesn't seem to be in the [editorconfig spec](https://editorconfig-specification.readthedocs.io/) nor in the [Prettier API](https://prettier.io/docs/en/api.html)

File: starters/solidjs-tanstack-tailwind/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "trailingComma": "es5",
```
Comment (mbicknese) [inline starters/solidjs-tanstack-tailwind/.prettierrc:2]: This is the default value and can be omitted.

**Review by mbicknese — CHANGES_REQUESTED** [review]
Love that you added an `.editorconfig` file!

Found some small things we could touch up. If you want to keep the `quote_type` rule in there, could you also apply it to '.js' files?

File: starters/solidjs-tanstack-tailwind/.editorconfig (hunk 2 above)
Comment (hdJerry) [reply starters/solidjs-tanstack-tailwind/.editorconfig:12]: I think at the moment it's a proposal.
So, should we take it out?
@mbicknese 

Screenshot 2022-10-18 at 17 46 58

File: starters/solidjs-tanstack-tailwind/.prettierrc (hunk 3 above)
Comment (hdJerry) [reply starters/solidjs-tanstack-tailwind/.prettierrc:2]: Thanks!!

File: starters/solidjs-tanstack-tailwind/.editorconfig (hunk 2 above)
Comment (mbicknese) [reply starters/solidjs-tanstack-tailwind/.editorconfig:13]: That's pretty cool, let's roll with it. 👍

Comment (hdJerry) [reply starters/solidjs-tanstack-tailwind/.editorconfig:13]: Here is the link for the screenshot.
https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties

**Pull request merged**